src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -1422,22 +1422,24 @@ private void ExecuteUninstallPackageCommand(object sender, ExecutedRoutedEventAr
                 return;
             }
 
-            UninstallPackage(package.Id);
+            UninstallPackage(package.Id, new[] { package });
         }
 
-        private void SetOptions(NuGetUI nugetUi, NuGetActionType actionType)
+        private void SetOptions(NuGetUI nugetUi, NuGetActionType actionType, IEnumerable<PackageItemViewModel> packages)
         {
             var options = _detailModel.Options;
+            IEnumerable<PackageItemViewModel> vulnerablePkgs = packages?.Where(x => x.Vulnerabilities.Any());
 
             nugetUi.FileConflictAction = options.SelectedFileConflictAction.Action;
             nugetUi.DependencyBehavior = options.SelectedDependencyBehavior.Behavior;
             nugetUi.RemoveDependencies = options.RemoveDependencies;
             nugetUi.ForceRemove = options.ForceRemove;
             nugetUi.DisplayPreviewWindow = options.ShowPreviewWindow;
             nugetUi.DisplayDeprecatedFrameworkWindow = options.ShowDeprecatedFrameworkWindow;
-
             nugetUi.Projects = Model.Context.Projects;
             nugetUi.ProjectContext.ActionType = actionType;
+            nugetUi.TopLevelVulnerablePackagesCount = vulnerablePkgs?.Count() ?? 0;
+            nugetUi.TopLevelVulnerablePackagesMaxSeverity = nugetUi.TopLevelVulnerablePackagesCount > 0 ? vulnerablePkgs.Max(pkgItem => pkgItem.Vulnerabilities?.Count() ?? -1) : -1;
```

Comment:
Added.

Reply:
@jebriede was chatting with dominoFire and it's not clear why we would need a list of vulnerabilities.
Can you help us understand with a simple example?

### Sample

Given:
> Updated Packages: [Newtonsoft.JSON; Chakra.Core; NuGet.Core]
> Vulnerabilities: [1, 3]

`TopLevelVulnerablePackagesCount`: **2**
Original `TopLevelVulnerablePackagesMaxSeverity`: **3**
Proposed "List of" `TopLevelVulnerablePackagesMaxSeverity`: **[1,3]**

We already know there are 2 vulnerable packages in this action. What can we do with the 1?
It can become obscure given we don't have a mapping here of _which_ package is the 1 or the 3.